### PR TITLE
fix(StatusIcon): add aria label to icons

### LINKED ISSUE #764: Bug: StatusIcon should allow for arria label on icon.
## Bug - 

StatusIcon should allow for aria-label on icon. 

**Actual behavior -**

tatusIcon does not allow for arria label on icon. 

### Steps for reproducing

no prop for iconLabel

### PR DESCRIPTION
**Pull request by Deriohig**
## Affected issues

- Resolves #764 

## Proposed changes

- adds an aria label to the icon to allow for status accessibility 
https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html

## Testing instructions

- check preview knob and inspect the element

### CODE REVIEW

File: src/components/StatusIcon/StatusIcon.js [hunk 1]
```diff
@@ -37,13 +37,17 @@ export default class StatusIcon extends Component {
 
     /** @type {string} Status. */
     status: oneOf(STATUS),
+
+    /** @type {string} icon aria label. */
+    iconLabel: string,
```
Comment (SimonFinney) [inline src/components/StatusIcon/StatusIcon.js:42]: Would it be possible to call this `iconDescription` to maintain consistency with Carbon? https://github.com/carbon-design-system/carbon/search?q=iconDescription

Comment (Deriohig) [reply src/components/StatusIcon/StatusIcon.js:42]: Sure thing!

**Comment by Deriohig** [comment]
@SimonFinney updated

**Review by SimonFinney — APPROVED** [review]
LGTM! 🔥

**Pull request merged**